train.py

```diff
@@ -110,323 +137,163 @@
-                        # set the "truly masked" values to unmasked, so they also get
-                        # ignored in the loss
-                        mask[real_mask] = False
-                        loss = mse(y_pred[mask], y[mask])
-                        current_batch_size = len(x)
-                        total_eo_val_loss += loss.item()
-                        num_val_updates_captured += 1
-
-                # ------------------------ Metrics + Logging -------------------------------
-                # train_loss now reflects the value against which we calculate gradients
-                train_eo_loss = total_eo_train_loss / num_updates_being_captured
-                val_eo_loss = total_eo_val_loss / num_val_updates_captured
-
-                if "train_size" not in training_config and "val_size" not in training_config:
-                    training_config["train_size"] = train_size
-                    training_config["val_size"] = val_size
-                    if wandb_enabled:
-                        wandb.config.update(training_config)
-
-                to_log = {
-                    "train_eo_loss": train_eo_loss,
-                    "val_eo_loss": val_eo_loss,
-                    "training_step": training_step,
-                    "epoch": epoch,
-                    "lr": lr,
-                }
-                tqdm_epoch.set_postfix(loss=val_eo_loss)
-
-                val_task_results, _ = validation_task.finetuning_results(
-                    model, sklearn_model_modes=["Random Forest"]
-                )
-                to_log.update(val_task_results)
-
-                if lowest_validation_loss is None or val_eo_loss < lowest_validation_loss:
-                    lowest_validation_loss = val_eo_loss
-                    best_val_epoch = epoch
-
-                    model_path = model_logging_dir / Path("models")
-                    model_path.mkdir(exist_ok=True, parents=True)
-
-                    best_model_path = model_path / f"{model_name}{epoch}.pt"
-                    logger.info(f"Saving best model to: {best_model_path}")
-                    torch.save(model.state_dict(), best_model_path)
-
-                # reset training logging
-                total_eo_train_loss = 0.0
-                num_updates_being_captured = 0
-                train_size = 0
-                num_validations += 1
-
-                if wandb_enabled:
-                    wandb.log(to_log)
-
-                model.train()
-
-logger.info(f"Trained for {num_epochs} epochs, best model at {best_model_path}")
-
-if best_model_path is not None:
-    logger.info("Loading best model: %s" % best_model_path)
-    best_model = torch.load(best_model_path, map_location=device)
-    model.load_state_dict(best_model)
-else:
-    logger.info("Running eval with randomly init weights")
+if task_type == "croptype":
+    df = WorldCerealBase.map_croptypes(df, finetune_classes, downstream_classes)
+    df = filter_remove_noncrops(df)
```

Comment:
see earlier comment. I think we kind of weaken the crop type classifier here because we only focus on `11`. Shouldn't we also add crops from the `12` class and maybe also some others, to make the crop set even more diverse? I think at some point I was even including grass in order for the crop type model to still be able to work with errors coming out o the crop/no-crop classifier. TBD!